hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/main/java/org/apache/hadoop/yarn/client/cli/RMAdminCLI.java

```diff
@@ -757,6 +768,28 @@ private int replaceLabelsOnNodes(Map<NodeId, Set<String>> map,
     return 0;
   }
 
+  private int failover(CommandLine cmd)
+      throws IOException, ServiceFailedException {
+    int numOpts = cmd.getOptions() == null ? 0 : cmd.getOptions().length;
+    final String[] args = cmd.getArgs();
+    if (numOpts > 1 || args.length != 2) {
+      errOut.println("failover: incorrect arguments");
+      printUsage(errOut, "-failover", YARN_HA_USAGE);
+      return -1;
+    }
+    HAServiceTarget fromNode = resolveTarget(args[0]);
+    HAServiceTarget toNode = resolveTarget(args[1]);
+    setRequestSource(HAServiceProtocol.RequestSource.REQUEST_BY_USER_FORCED);
+    HAServiceProtocol proto = fromNode.getProxy(
+            getConf(), 0);
+    HAServiceProtocolHelper.transitionToStandby(proto, createReqInfo());
+    HAServiceProtocol proto1 = toNode.getProxy(
+            getConf(), 0);
+    HAServiceProtocolHelper.transitionToActive(proto1, createReqInfo());
```

Comment:
Do we need to check whether args[1] is the primary? If it is the primary, there is no need to switch.

Reply:
If  args[1] is the primary,it will return ：
HAServiceProtocolHelper.transitionToActive ->AdminService.java(transitionToActive):
"
  @Override
  public synchronized void transitionToActive(
      HAServiceProtocol.StateChangeRequestInfo reqInfo) throws IOException {
    if (isRMActive()) {
      **return;**
    }
....
"